src/core_atmosphere/mpas_atm_halos.F

```diff
@@ -0,0 +1,446 @@
+! Copyright (c) 2013,  Los Alamos National Security, LLC (LANS)
+! and the University Corporation for Atmospheric Research (UCAR).
+!
+! Unless noted otherwise source code is licensed under the BSD license.
+! Additional copyright and license information can be found in the LICENSE file
+! distributed with this code, or at http://mpas-dev.github.com/license.html
+!
+module mpas_atm_halos
+
+   use mpas_derived_types
+   use mpas_pool_routines
+   use mpas_log, only : mpas_log_write, mpas_log_info
+
+   !
+   ! Abstract interface for routine used to communicate halos of fields
+   ! in a named group
+   !
+   abstract interface
+      subroutine halo_exchange_routine(domain, halo_group, ierr)
+
+         use mpas_derived_types, only : domain_type
+
+         type (domain_type), intent(inout) :: domain
+         character(len=*), intent(in) :: halo_group
+         integer, intent(out), optional :: ierr
+
+      end subroutine halo_exchange_routine
+   end interface
+
+   character(len=StrKIND), pointer, private :: config_halo_exch_method
+   procedure (halo_exchange_routine), pointer :: exchange_halo_group
+
+
+   contains
+
+
+   !-----------------------------------------------------------------------
+   !  routine atm_build_halo_groups
+   !
+   !> \brief Builds halo exchange groups used throughout atmosphere core
+   !> \author Michael Duda
+   !> \date   5 June 2023
+   !> \details
+   !>  This routine builds the halo exchange groups that are used throughout
+   !>  the atmosphere core, and it sets a function pointer,
+   !>  exchange_halo_group, to the routine that may be used to exchange the
+   !>  halos for all fields in a named group.
+   !>
+   !>  A value of 0 is returned if halo exchange groups have been
+   !>  successfully set up and a non-zero value is returned otherwise.
+   !
+   !-----------------------------------------------------------------------
+   subroutine atm_build_halo_groups(domain, ierr)
+
+      use mpas_dmpar, only : mpas_dmpar_exch_group_create, mpas_dmpar_exch_group_add_field, &
+	                     mpas_dmpar_exch_group_full_halo_exch 
```

Comment:
This line generates the following warning from the gfortran compiler with `-Wall -std=f2008`:
```
mpas_atm_halos.F:56:2:

   56 |                              mpas_dmpar_exch_group_full_halo_exch
      |         1
Warning: Nonconforming tab character at (1) [-Wtabs]
```